src/vaadin-message-input.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * @license
+ * Copyright (c) 2021 Vaadin Ltd.
+ * This program is available under Apache License Version 2.0, available at https://vaadin.com/license/
+ */
+import { PolymerElement, html } from '@polymer/polymer/polymer-element.js';
+import { ThemableMixin } from '@vaadin/vaadin-themable-mixin/vaadin-themable-mixin.js';
+import { ElementMixin } from '@vaadin/vaadin-element-mixin/vaadin-element-mixin.js';
+import '@vaadin/vaadin-button/src/vaadin-button.js';
+import '@vaadin/vaadin-text-field/src/vaadin-text-area.js';
+/**
+ * `<vaadin-message-input>` is a Web Component for sending messages.
+ * It consists of a text area that grows on along with the content, and a send button to send message.
+ *
+ * ```html
+ * <vaadin-message-input></vaadin-message-input>
+ * ```
+ *
+ * ### Styling
+ *
+ * The following shadow DOM parts are available for styling:
+ *
+ * Part name | Description
+ * ----------|----------------
+ *
+ * See [ThemableMixin – how to apply styles for shadow parts](https://github.com/vaadin/vaadin-themable-mixin/wiki)
+ *
+ * @extends HTMLElement
+ * @mixes ThemableMixin
+ * @mixes ElementMixin
+ */
+class MessageInputElement extends ElementMixin(ThemableMixin(PolymerElement)) {
+  static get template() {
+    return html`
+      <style>
+        :host {
+          align-items: flex-start;
+          box-sizing: border-box;
+          display: flex;
+          max-height: 50vh;
+          overflow: hidden;
+        }
+        vaadin-text-area {
+          align-self: stretch;
+          flex-grow: 1;
+          padding: 0;
+        }
+        vaadin-button {
+          flex-shrink: 0;
+          margin: 0;
+        }
+      </style>
+      <vaadin-text-area placeholder="Message"></vaadin-text-area>
+      <vaadin-button theme="primary contained">Send</vaadin-button>
+    `;
+  }
+
+  ready() {
+    super.ready();
+
+    // Set aria-label to provide an accessible name for the labelless input
+    const textarea = this.shadowRoot.querySelector('vaadin-text-area').inputElement;
```

Comment:
Just pointing out that with the "slotted input" feature we could set the attributes and styles for `textarea` in the template instead of in JS (as discussed in Slack). Still not sure if that's the best way, so not blocking.

Reply:
Create an own input and slot it into vaadin-text-area, to be able to set the labels on it? I think it is cleaner this way.